common.md

```diff
@@ -492,6 +494,20 @@ We recursively convert the contents of pairs, ors and options, if applicable.
   rule #MichelineToNative(Right V, or _:AnnotationList _:Type TR:Type, KnownAddrs, BigMaps) => Right #MichelineToNative(V, TR, KnownAddrs, BigMaps)
 ```
 
+Externally, tickets are represented by `Pair`-constructors.
+Internally, we represent tickets with the `#Ticket`-constructor.
+Notice that this rule does not overlap with the rule for ordinary pairs
+since they match on different types (second argument of `#MichelineToNative`).
+
+```k
+  rule #MichelineToNative(Pair Addr (Pair V N), ticket _:AnnotationList T:Type, KnownAddrs, BigMaps) =>
+    #Ticket(
+      #MichelineToNative(Addr, address .AnnotationList, KnownAddrs, BigMaps),
+      #MichelineToNative(V, T, KnownAddrs, BigMaps),
+      #MichelineToNative(N, nat .AnnotationList, KnownAddrs, BigMaps)
```

Comment:
Because we changed the type of tickets to represent values more precisely, we now need to update the internal representation conversion function by appropriately downcasting the first and third arguments. Given the definition of `#MichelineToNative`, we know that the downcast will always succeed for tickets that are well-formed in the external representation.

```suggestion
      {#MichelineToNative(Addr, address .AnnotationList, KnownAddrs, BigMaps)}:>Address,
      #MichelineToNative(V, T, KnownAddrs, BigMaps),
      {#MichelineToNative(N, nat .AnnotationList, KnownAddrs, BigMaps)}:>Int
```

Note that a similar trick is used on lines 589 and 597, for example.